MovieQuiz/Services/StatisticService.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+
+protocol StatisticService {
+    var totalAccuracy: Double { get }
+    var gamesCount: Int { get }
+    var bestGame: GameRecord { get }
+    func store(correct count: Int, total amount: Int)
+}
+final class StatisticServiceImplementation: StatisticService {
+    
+    private let userDefaults = UserDefaults.standard
+    
+    private enum Keys: String {
+        case correct, total, bestGame, gamesCount
+    }
+    
+    
+    private(set) var totalAccuracy: Double {
+        get {
+            guard let data = userDefaults.data(forKey: Keys.total.rawValue),
+                  let record = try? JSONDecoder().decode(Double.self, from: data) else {
+                return 0
+            }
```

Comment:
UserDefaults поддерживает работу с примитивными типами [для чтения](https://developer.apple.com/documentation/foundation/userdefaults/1407405-integer) и [записи](https://developer.apple.com/documentation/foundation/userdefaults/1413614-set).
Совсем не обязательно простой тип гонять через декодер, когда есть простые способы данные получить/записать.

Reply:
Есть простые и проще, но в спринте указан именно этот(

Reply:
Согласен и так же подумал, но у нас дано перечисление, для данных через декодер, тогда зачем мы его указываем?
```suggestion
            private enum Keys: String {
        case correct, total, bestGame, gamesCount
    }
```

Reply:
исправлено